server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -1048,17 +1049,26 @@ public void snapshotShard(Store store, MapperService mapperService, SnapshotId s
             final GroupedActionListener<Void> filesListener =
                 new GroupedActionListener<>(allFilesUploadedListener, indexIncrementalFileCount);
             final Executor executor = threadPool.executor(ThreadPool.Names.SNAPSHOT);
+            final AtomicBoolean alreadyFailed = new AtomicBoolean();
```

Comment:
Can you add a small comment which explains why we need this flag?

Reply:
Done :)